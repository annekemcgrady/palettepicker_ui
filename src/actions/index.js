export const getColors = colors => ({
    type: 'ADD_COLORS', 
    colors
})

export const lockColor = color => ({
    type: 'LOCK_COLOR',
    color
})

export const getProjects = projects => ({
    type: 'GET_PROJECTS',
    projects
})

export const getPalettes = palettes => ({
    type: 'GET_PALETTES',
    palettes
})

export const addProject = project => ({
    type: 'ADD_PROJECT',
    project
})

export const addPalette = palette => ({
    type: 'ADD_PALETTE',
    palette
})

export const loadComplete = () => ({
    type: 'LOAD_COMPLETE'
})

export const hasErrored = error => ({
    type: 'HAS_ERRORED',
    error
})