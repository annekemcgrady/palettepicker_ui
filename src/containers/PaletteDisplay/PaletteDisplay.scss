.palette-display {

  width: 100%;
  height: 50vh;
  background-color: aliceblue;
  display: flex;
  flex-direction: column
}

.palette-tiles {
  display: flex;
}

.generate-new-palette-button {
  width: 150px;
  height: 40px;
  align-self: center;
}