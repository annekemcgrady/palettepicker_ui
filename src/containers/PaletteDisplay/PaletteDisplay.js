import React, { Component } from "react";
import { connect } from "react-redux";
import { getColors } from "../../actions";
import PaletteTile from "../../components/PaletteTile/PaletteTile.js";
import PaletteForm from '../PaletteForm/PaletteForm';
import "./PaletteDisplay.scss";
import ColorScheme from "color-scheme";

export class PaletteDisplay extends Component {
    constructor() {
    super()
    this.state = {
        colors: []
    }
}
    generateColors = () => {
        let randomHue = Math.floor(Math.random() * 10000 + 1);
        let scheme = new ColorScheme();
        scheme
        .from_hue(randomHue)
        .scheme("contrast")
        .variation("hard");


        let colors = scheme.colors().splice(0, 5);
        let colorObjects = colors.map(color => {
            return {hexCode: `#${color}`, isLocked: false}
        })

        const evaluatedColors =  this.props.colors.map((color, index) => {
            return color.isLocked ? color : colorObjects[index]
          })
          if(!evaluatedColors.length) {
              this.props.setColors(colorObjects)
          } else {
              this.props.setColors(evaluatedColors);
          }
    };

    lockColor = (id) => {
        const lockedColors = this.props.colors.map(tile => {
            if(tile.hexCode === id) {
            tile.isLocked = !tile.isLocked
            }
            return tile
        })
        this.props.setColors(lockedColors)
    }

    render = () => {
    const displaySwatches = this.props.colors.map((color, index) => {
        return (
            <PaletteTile 
                key={'pal-' + index} 
                id={color.hexCode} 
                isLocked={color.isLocked} 
                hexCode={color.hexCode.toUpperCase()} 
                lockColor={this.lockColor} 
            />
        );
    });

    return (
        <section className="palette-display">
            <div className='palette-tiles'>{displaySwatches}</div>
            <button
                type="button"
                className="generate-new-palette-button"
                onClick={this.generateColors}
                >
                Generate New Palette
            </button>
          <PaletteForm />
        </section>
    );
    };

}


export const mapStateToProps = state => ({
    colors: state.colors
});

export const mapDispatchToProps = dispatch => ({
    setColors: colors => dispatch(getColors(colors))
})


export default connect(
    mapStateToProps, mapDispatchToProps)(PaletteDisplay);
