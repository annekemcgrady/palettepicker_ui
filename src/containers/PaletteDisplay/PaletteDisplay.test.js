import React from 'react';
import { PaletteDisplay, mapStateToProps, mapDispatchToProps} from './PaletteDisplay';
import { getColors } from '../../actions';
import { shallow } from 'enzyme';


describe('PaletteDisplay', () => {
  let wrapper;
  let instance;
  let props;


  beforeEach(() => {
  props = {
    colors: [{hexCode:'#123457', isLocked: false}, {hexCode:'#123456', isLocked: false}, {hexCode:'#123456', isLocked: false}, {hexCode:'#123456', isLocked: false}, {hexCode:'#123456', isLocked: false}],
    setColors: jest.fn()
  };

  wrapper = shallow(<PaletteDisplay{...props}/>);
  instance = wrapper.instance();

})

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();

  })

 describe('generateColors', () => {
  it.skip('should call generateColors when Generate New Colors button clicked', ()=> {
      
      instance.generateColors = jest.fn();
      wrapper.find('.generate-new-palette-button').simulate('click')
      expect(instance.generateColors).toHaveBeenCalled()
    })

  it('should dispatch a setColors action with an array of five hexcodes', ()=> {

  })

})

 describe('lockColor', ()=> {

  it('should call setColors with a new array of colors, ()=> {
    let expected = [{hexCode:'#123457', isLocked: true}, {hexCode:'#123456', isLocked: false}, {hexCode:'#123456', isLocked: false}, {hexCode:'#123456', isLocked: false}, {hexCode:'#123456', isLocked: false}]
    instance.lockColor('#123457')
    const mockDispatch = jest.fn();
    expect(mockDispatch).toHaveBeenCalledWith(expected)
  })

  it('should dispatch a setColors action with an array of five hexcodes', ()=> {

  })

  })

  describe('mapStateToProps', ()=> {

    it('should return colors from store', () => {
      const mockState = {
          colors: [{color: '123456'}, {color: 'ffffff'}]
      }

      const expected = {
          colors: [{color: '123456'}, {color: 'ffffff'}]
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
  });

  })

  describe('mapDispatchToProps', ()=> {

    it('should call dispatch with a setColors action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getColors([{hexCode: '#123ABC'}]);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setColors([{hexCode: '#123ABC'}])
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })

})