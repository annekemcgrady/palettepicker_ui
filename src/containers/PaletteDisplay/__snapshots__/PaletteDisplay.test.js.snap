// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteDisplay should match the snapshot 1`] = `ShallowWrapper {}`;
